core/src/main/java/com/aem/demo/core/services/impl/LikeDislikeServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package com.aem.demo.core.services.impl;
+
+import com.aem.demo.core.services.LikeDislikeService;
+import com.aem.demo.core.utils.ResolverUtil;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.sling.api.resource.*;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+@Slf4j
+@Component(service = LikeDislikeService.class, immediate = true)
+public class LikeDislikeServiceImpl implements LikeDislikeService {
+
+    @Reference
+    private ResourceResolverFactory resourceResolverFactory;
+
+    @Override
+    public int updateLikeDislike(String property, String rootPath) {
+        int incrementProperty;
+        int result = 0;
+        try (ResourceResolver resolver = ResolverUtil.newResolver(resourceResolverFactory)) {
+            Resource resource = resolver.getResource(rootPath);
+            ModifiableValueMap properties = resource.adaptTo(ModifiableValueMap.class);
+            incrementProperty = properties.get(property, 0);
+            properties.put(property, incrementProperty + 1);
+            result = properties.get(property, 0);
```

Comment:
Generally, you can just assign `result = incrementProperty + 1` without re-reading the property from the resource. Reading from the resource is a lengthier routine. 

But, then again, this piece of code is generally neat and clean, thank you!)

P.S. I just wrote "result = incrementProperty + 1".  I just reflected your code. It doesn't mean this code could not be better. Why not: 
```
result = properties.get(property, 0) + 1;
properties.put(property, result);
resolver.commit();
// ...
return result;
```
All the same in just two lines of code,..

But there's not a mistake in this code, anyway